pkg/k8s/configmaps/crud_test.go

```diff
@@ -0,0 +1,146 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmaps_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	apiv1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	k8sTesting "k8s.io/client-go/testing"
+)
+
+func TestGet(t *testing.T) {
+	ctx := context.Background()
+	ns := "fake-ns"
+	mockCfgName := "fake-cfg"
+	mockCfgStatus := "fake-status"
+	mockCfg := &apiv1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mockCfgName,
+			Namespace: ns,
+		},
+		Data: map[string]string{
+			"status": mockCfgStatus,
+		},
+	}
+	clientset := fake.NewSimpleClientset(mockCfg)
+
+	cfg, err := configmaps.Get(ctx, mockCfgName, ns, clientset)
+	if err != nil {
+		t.Fatalf("unexpected error returned by Get(): %v", err)
+	}
+
+	if mockCfgName != cfg.Name {
+		t.Errorf("Expected cfg name %s, got %s", mockCfgName, cfg.Name)
+	}
+
+	if mockCfgStatus != cfg.Data["status"] {
+		t.Errorf("Expected cfg status %s, got %s", mockCfgStatus, cfg.Data["status"])
+	}
+}
+
+func TestGet_Err(t *testing.T) {
+	ctx := context.Background()
+
+	mockErr := errors.New("unexpected error")
+
+	clientset := fake.NewSimpleClientset()
+	clientset.Fake.PrependReactor("get", "configmaps", func(action k8sTesting.Action) (handled bool, ret runtime.Object, err error) {
+		return true, nil, mockErr
+	})
+
+	cfg, err := configmaps.Get(ctx, "fake-cfg-name", "fake-ns", clientset)
+	if cfg != nil {
+		t.Errorf("Expected cfg to be nil, but got: %v", cfg)
+	}
+
+	if err == nil || err.Error() != mockErr.Error() {
+		t.Errorf("Expected error to be '%v', but got: %v", mockErr, err)
+	}
+}
+
+func TestWaitForStatus(t *testing.T) {
+	ctx := context.Background()
+
+	mockCfgName := "fake-cfg"
+	ns := "fake-ns"
+
+	mockCfg := &apiv1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mockCfgName,
+			Namespace: ns,
+		},
+		Data: map[string]string{
+			"status": "fake-initial-status",
+		},
+	}
+
+	clientset := fake.NewSimpleClientset(mockCfg)
+
+	go func() {
+		mockCfg.Data["status"] = "fake-final-status"
+		clientset.CoreV1().ConfigMaps(ns).Update(ctx, mockCfg, metav1.UpdateOptions{})
+	}()
+
+	err := configmaps.WaitForStatus(ctx, mockCfgName, ns, "fake-final-status", 5*time.Second, clientset)
```

Comment:
I will have a look at this PR today, for now if you agree we can merge as is and come back to it later once we agreed upon an approach?

Reply:
@jLopezbarb I made a small change but now the tests run super fast, because I am injecting the ticker, which I think it's a nice thing to do anyway, so from the outside the function, it can be controlled how frequently we check of a status change. Have a look at the diff: https://github.com/okteto/okteto/compare/b6a16882a4bbb4064252df190bfb3502c3739983..ceafa44661010c21ecb57b94142aeb5ca6eab6d2

This also removes the issue of running the tests "slowly"